packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1694,24 +1684,23 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
     }
 
     /**
-     * Run async callback with given timeout and retry.
+     * Run async callback with given timeout and retry. Returns the result of the callback.
      * @ignore
      */
-    protected async _timeoutAndRetry(
-        handler: () => Promise<unknown>,
+    protected async _timeoutAndRetry<T>(
+        handler: () => Promise<T>,
         timeout: number,
         error: Error | string,
         maxRetries = 3,
         retried = 1,
-    ): Promise<void> {
+    ): Promise<T> {
         try {
-            await addTimeoutToPromise(handler, timeout, error);
+            return addTimeoutToPromise(handler, timeout, error);
```

Comment:
Wait a second, if you return the promise and it throws, I don't think the catch branch here will _catch_ it, right?